cmake/FirstCuda.cmake

```diff
@@ -0,0 +1,233 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
```

Comment:
What did this file used to be?

Reply:
this file look like `mshadow\cmake\Cuda.cmake` but it remove some content of conflict.
besides, I made a mistake when I created `mshadow\cmake\Cuda.cmake`. It shouldn't have been put `mshadow\cmake\`.should be placed on `root\cmake`

Reply:
Is there a way to modify the mshadow Cuda.cmake so that we donlt have several versions of almost-the-same files floating around in the cmake include paths?  Sounds like a confusing and hard-to-debug build problem waiting to happen.

Reply:
Considering that mshadow will be removed gradually,So consider moving `mshadow\cmake` to `root\cmake` gradually.Do you mean to do it all at once?

Reply:
Or give them different file names...